primitives/src/v2/mod.rs

```diff
@@ -1569,6 +1594,72 @@ impl CompactStatement {
 	}
 }
 
+/// `IndexedVec` struct indexed by type specific indexes.
+#[derive(Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(PartialEq, MallocSizeOf))]
+pub struct IndexedVec<K, V>(Vec<V>, PhantomData<fn(K) -> K>);
```

Comment:
Could someone confirm this has the same SCALE (d)encoding as a regular `Vec`?

Reply:
Yes it has the same. `PhantomData` encodes into "nothing" and decodes from "nothing"